### ISSUE TITLE
Large memory leak with getPacket buffer

### ISSUE THREAD

**Issue by andrewbenton** [mysql-d/mysql-native#82]
I'm finding that even after a connection and all statements have been explicitly `close()`'d and a GC cycle run with `GC.collect`, I'm seeing about 1/2 of the buffers from getPacket stuck in memory.

**Comment by Abscissa** [comment]
I'm not certain how to check this. How are you determining getPacket's buffers aren't being collected? Not trying to pass the buck or anything, but have you looked into whether you might have some other data generating false pointers? Is this on 64-bit?

I can definitely reduce the amount of allocations in the first place, but if something in mysql-native is holding onto these old buffers, then I definitely want to fix that regardless.

**Comment by Abscissa** [comment]
One thing to be aware of (that I just realized while digging through it): When mysql-native returns a string (and I think binary data too), it returns a slice directly from getPacket's buffer. So if you still have string/binary data live in your program, that originated from mysql, then that would be getPacket data that can't/shouldn't be garbage collected.

**Comment by andrewbenton** [comment]
The way I can tell it's memory from getPacket is that the memory profiler in DMD indicates that's the only thing that has allocated enough memory to be causing a problem.  It looks like some of the memory from getPacket is being collected, but not all of it.  I tried throwing an idup on all of the strings I was getting out, but that didn't change consumption.

It's possible that ddbc is keeping an extra pointer around, since that's how I'm using mysql-native, but when I looked at their API, it seemed they were just providing an interface down to mysql-native.  This is on 64-bit linux, and I see the memory issue with both dmd and ldc.
